### PR TITLE
time format in 24 hours

### PR DESCRIPTION
**Pull request by sonujangra27**
### What do you want to achieve?
time format in 24 hours
-

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [x] I have compared it with mocks and all design elements are the same.
- [] I have tested it in IE Browser.

### How was the code tested?

- add timeFormat24Hours: True in install script to test it

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/mck-app.js [hunk 1]
```diff
@@ -468,6 +468,10 @@ function ApplozicSidebox() {
                 options.voiceNote != null
                     ? options.voiceNote
                     : widgetSettings && widgetSettings.voiceNote;
+            options.timeFormat24Hours =
+                options.timeFormat24Hours != null
+                    ? options.timeFormat24Hours
+                    : widgetSettings;          
             KommunicateUtils.deleteDataFromKmSession('settings');
```
Comment (Prochnost55) [inline webplugin/js/app/mck-app.js:475]: what if `options.timeFormat24Hours` is null?

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 2]
```diff
@@ -9628,6 +9627,28 @@ var userOverride = {
...
             };
 
+            
+                _this.getMessageCreatedAtTime=function(createdAtTime){
+                    if(TIME_FORMAT_24HOURS){
+                    var msgtime = new Date(createdAtTime);
+                    function addZero(i) {
+                        if (i < 10) {i = "0" + i}
+                        return i;
+                    }
```
Comment (Prochnost55) [inline webplugin/js/app/mck-sidebox-1.0.js:9637]: Please correct the formatting.

**Review by Prochnost55 — CHANGES_REQUESTED** [review]

**Comment by Prochnost55** [comment]
@sonujangra27 any updates?

File: webplugin/js/app/mck-app.js (hunk 1 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-app.js:475]: done

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 2 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js:9637]: done

**Review by Prochnost55 — APPROVED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 3]
```diff
@@ -9628,6 +9627,27 @@ var userOverride = {
                 return '';
             };
 
+            
+             _this.getMessageCreatedAtTime=function(createdAtTime){
+                 if(TIME_FORMAT_24HOURS){
+                    var msgtime = new Date(createdAtTime);
+                    function addZero(i) {
+                        if (i < 10) {i = "0" + i}
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:9635]: The variables declared without the var keyword becomes global variables
it should be declared

@Prochnost55 can confirm

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

**Review by devashishmamgain — APPROVED** [review]

**Review by Prochnost55 — APPROVED** [review]

**Review by Prochnost55 — CHANGES_REQUESTED** [review]
Screenshot 2022-02-03 at 11 04 06 AM
Screenshot 2022-02-03 at 11 03 30 AM

Time is comming but I think date should also be coming
@devashishmamgain

**Comment by devashishmamgain** [comment]
> Screenshot 2022-02-03 at 11 04 06 AM Screenshot 2022-02-03 at 11 03 30 AM
> 
> Time is comming but I think date should also be coming @devashishmamgain

yes date need to come
@sonujangra27

**Comment by Prochnost55** [comment]
@sonujangra27 any updates?

**Comment by sonujangra27** [comment]
Made changes!
Added date.

Sonu Jangra
Technical Support Engineer

<https://www.kommunicate.io/>

  (+1) (310) 402-2374

  kommunicate.io

<https://www.facebook.com/kommunicateio> <https://twitter.com/kommunicateio>
<https://www.linkedin.com/company/kommunicate/>
<https://www.youtube.com/channel/UCUZbGHJ7kOA0A2bY_OFkGHQ>

<https://www.kommunicate.io/blog/category/product-updates/>

**Review by Prochnost55 — DISMISSED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 4]
```diff
@@ -9628,6 +9627,36 @@ var userOverride = {
                 return '';
             };
 
+            
+            _this.getMessageCreatedAtTime = function (createdAtTime) {
+                if (TIME_FORMAT_24HOURS) {
+                    let month = ["Jan","Feb","March","April","May","June","July","Aug","Sep","Oct","Nov","Dec"];
```
[the remark this answers is not in the document]
Comment (devashishmamgain) [reply webplugin/js/app/mck-sidebox-1.0.js:9633]: some place it is full name March
at other places it is first 3 characters Jan
what is the standard pattern used across the world?

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 5]
```diff
@@ -9628,6 +9627,36 @@ var userOverride = {
...
+                        } else {
+                            return i;
+                        }
+                    }
+                    var hours = addZero(msgtime.getHours());
+                    var minute = addZero(msgtime.getMinutes());
+                    if(msgtime.getHours()<24){
+                        var time =   hours + ":" + minute;
+                    }else{
+                        var time = monthName + " " + messageDay + ", " + hours + ":" + minute;
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:9649]: as it used below, it can't be a local variable inside if else

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 5 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js]: made changes

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 6]
```diff
@@ -9628,6 +9627,35 @@ var userOverride = {
...
             };
 
+            
+            _this.getMessageCreatedAtTime = function (createdAtTime) {
+                if (TIME_FORMAT_24HOURS) {
+                    
+                    var month = ["Jan", "Feb", "Mar", "April", "May", "June", "July", "Aug", "Sep", "Oct", "Nov", "Dec"];
+                    var msgtime = new Date(createdAtTime);
+                    var messageDay = msgtime.getDate(createdAtTime);
+                    var monthName = month[msgtime.getMonth(createdAtTime)];
```
Comment (Prochnost55) [inline webplugin/js/app/mck-sidebox-1.0.js:9637]: do we need to pass createdAtTime in getDate() and getMonth()?

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 7]
```diff
@@ -9628,6 +9627,35 @@ var userOverride = {
                 return '';
             };
 
+            
+            _this.getMessageCreatedAtTime = function (createdAtTime) {
+                if (TIME_FORMAT_24HOURS) {
+                    
+                    var month = ["Jan", "Feb", "Mar", "April", "May", "June", "July", "Aug", "Sep", "Oct", "Nov", "Dec"];
+                    var msgtime = new Date(createdAtTime);
```
Comment (Prochnost55) [inline webplugin/js/app/mck-sidebox-1.0.js:9635]: has this comment been resolved?
https://github.com/Kommunicate-io/Kommunicate-Web-SDK/pull/833#discussion_r811773038

**Review by Prochnost55 — CHANGES_REQUESTED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js]: corrected. it should be Mar

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 7 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js:9635]: yes

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 6 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js]: removed

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 8]
```diff
@@ -9628,6 +9627,35 @@ var userOverride = {
                 return '';
             };
 
+            
+            _this.getMessageCreatedAtTime = function (createdAtTime) {
+                if (TIME_FORMAT_24HOURS) {
+                    
+                    var month = ["Jan", "Feb", "Mar", "April", "May", "June", "July", "Aug", "Sep", "Oct", "Nov", "Dec"];
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:9634]: do we need to hard code these? there might be some function in javascript to get the month name based on the month number
how will the localization work? do we handle translation to other language in existing format?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 8 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js]: Removed the hardcoding. and used toLocaleString to get the month name.

**Review by Prochnost55 — APPROVED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 9]
```diff
@@ -9628,6 +9627,37 @@ var userOverride = {
                 return '';
             };
 
+            
+            _this.getMessageCreatedAtTime = function (createdAtTime) {
+                if (TIME_FORMAT_24HOURS) {
+
```
Comment (devashishmamgain) [inline webplugin/js/app/mck-sidebox-1.0.js:9633]: remove empty lines

**Review by devashishmamgain — DISMISSED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 9 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js]: done

**Review by devashishmamgain — APPROVED** [review]

**Pull request closed without merging**